storehaus-algebra/src/main/scala/com/twitter/storehaus/algebra/ConcurrentHashMapMergeableStore.scala

```diff
@@ -24,11 +39,11 @@ class ConcurrentHashMapMergeableStore[K, V](map: JConcurrentHashMap[K, V])(
 
   override def merge(kv: (K, V)): Future[Option[V]] = {
     val (k, v) = kv
-    map.get(k) match {
-      case null =>
-        if (null == map.putIfAbsent(k, v)) Future.None
+    Option(map.get(k)) match {
```

Comment:
let's not do this. This is allocating where in the past we were avoiding allocations. This could slow down operations in a path that may be important for performance.